src/rosdiscover/config.py

```diff
@@ -57,14 +58,25 @@ def from_dict(cls, dict_: Mapping[str, Any]) -> 'Config':
             raise ValueError("expected 'sources' to be a list")
         if not isinstance(dict_['launches'], list):
             raise ValueError("expected 'launches' to be a list")
-
+        if isinstance(dict_['launches'][0], dict):
+            launch_args_provided = True
```

Comment:
This seems to suggest that if the first entry in launches is a YAML object (i.e., it provides args), then all entries that follow must also provide args? If so, that's probably being a little overly restrictive.

If the intention is to check that the items in `launches` are valid, you could do something like:
```python
if not all(type(d) in (dict, str) for d in dict_["launches"]):
# or if any(...) 
```

Reply:
fixed